### DISCUSSION TITLE
Is there any plugin/API to maintain fixture scope/cache across the multiple test runs

### DISCUSSION THREAD

**Discussion by KrupalJoshi** [pytest-dev/pytest#8772] [Ideas]
Hi,

I would like to know is there any way (maintain cache, API) to re-use the fixture (for same set/values of parameters) across multiple test runs. 
Purpose: To minimize time taking to run same fixture for same set/values of parameters while developing/debugging test case. 

e.g.: Developing/debugging Test case. This test case using fixture. At first time fixture will get executed and then test case is starting execution. For any reason error/exceptions occurs and test case execution/runs get terminated. Run/debug the same test case here fixture will get call and executed this will needs some time. Instead of executing fixture again, if there is any way to cache/skip the fixture execution and re-use the fixture executed in previous run, it will save the time.

**Comment by RonnyPfannschmidt** [comment]
this question needs more context

typically with the exit of pytest, the python values of the fixtures get lost as the process containing them goes away
unless there is extra detail on what your fixtures are, no suggestion can be made other than - its not supported at the python level itself

**Reply by KrupalJoshi** [reply]
High level purpose is re-use fixtures even after test case gets terminated/failed/assert occurs.

e.g. fixture1 perform some operations and provide object1 as a result. Test case is using fixture1 with same parameter values. For first time test case runs, fixture1 gets executed provided object1, but anyhow test case failed/assert and run gets stopped. Now second time run test case, fixture1 gets executed, provided object1. Here fixture1 will perform the same execution and provide same result as in first run. Instead of this, can we skip fixture1 execution and re-use object1 as it provided first run?

One approach I can think is use of pytest cache API to store resulted object, but it does not support any user defined object as a value.

Please let me know in case you need more details.

**Reply by RonnyPfannschmidt** [reply]
The basic answer the is ask the pytest cache for a file and use that to store own data 

As no detail use case was given i can't provide more to the point advise

**Reply by KrupalJoshi** [reply]
>>The basic answer the is ask the pytest cache for a file and use that to store own data
     Is this supported by pytest cache? If so, can you please share web link/documents?

>>As no detail use case was given I can't provide more to the point advise
     Isn't above mentioned example is useful?

**Reply by RonnyPfannschmidt** [reply]
?

**Reply by nicoddemus** [reply]
@KrupalJoshi take a look at https://docs.pytest.org/en/stable/cache.html, might be what you are looking for. 👍

**Reply by KrupalJoshi** [reply]
@nicoddemus Thanks for sharing this. I am looking to store user defined object (result of fixture execution) to store in config cache API. I think as per documents this is not supported.

**Reply by nicoddemus** [reply]
@KrupalJoshi not directly, but can't you convert the user defined object to/from JSON? Or use [pickle](https://docs.python.org/3/library/pickle.html)?

**Reply by KrupalJoshi** [reply]
Yes, that I have to check yet.
